components/d2l-work-to-do/d2l-work-to-do.js

```
@@ -493,6 +493,18 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
 		}
 	}
 
+	// !TODO Further investigate defect DE42208 to remove need for retry
+	async _performSirenActionWithRetry(token, action, fields, immediate, maxRetries = 1)  {
+		if (maxRetries) {
+			return performSirenAction(token, action, fields, immediate)
+				.catch(() => {
+					return this._performSirenActionWithRetry(token, action, fields, immediate, maxRetries - 1);
```

my gut feeling is that recursion is a weird way to solve this problem but I don't know how else it could be done - if you do this in a `while` loop and `await performSirenAction`, could that accomplish the same thing? if not, I'm okay with this too

---

`async _performSirenActionWithRetry(token, action, fields, immediate, maxRetries = 1)  {
	for(i = -1; i < maxRetries; i++) {
		try {
			return await performSirenAction(token, action, fields, immediate);
		} catch {
			continue;		
		}
	}
}`

An iterative solution will probably look something like this. Not sure which is better/more readable

---

I don't think recursion is bad here. Since we control the exit condition, the call stack won't get terribly big. Maybe @AntonBazhal has other concerns either way.

---

Recursion is fine. My concern with the code above is that we swallow an actual error when the retry limit is reached.

---

The new commit should reject the promise with last retry error. The failure reason for the retries will likely be the same so that should be sufficient. The other alternative being append the error message for each retry and reject with a list of errors